index.js

```diff
@@ -225,21 +228,88 @@ module.exports = {
         // If there is no tests/index.html, the function will reject
         return;
       });
 
       return RSVP.all([indexPromise, testIndexPromise]);
+      //return RSVP.all([indexPromise]);
     }.bind(this));
   },
 
   indexBuilder: function (config) {
+    //----------------------------------------------------------------------------
+    //STEPS:
+    //1 - handle the configFile if defined
+    //1 - get script names from index.html
+    //2 - cook the amd-start script as a string
+    //3 - if inline === true : inline the scripts
+    //  - if inline === false : add script tags to the amd-start/amd-config files
+    //----------------------------------------------------------------------------
     
-    // Load the index file    
+    // Full path to the index file    
+    var indexPath = path.join(config.directory, config.indexFile);
+
+    //1 - deal with the amd config file if defined
+    var amdConfigInfo = this.handleAmdConfig(config);
+
+    var self = this;
+    //2 - get the scripts from index.html
+    return this.getScriptsFromIndex(indexPath)
+      .then(function(scripts){
+        config.scriptsAsString = scripts;
+        //3- create the start script
+        var amdStartScriptInfo = self.startScriptBuilder(config);  
+        //if we are inlining...
+        if(config.inline){
+          config.amdConfig = amdConfigInfo.amdConfig;
+          config.amdStart = amdStartScriptInfo.amdStart;
+          return self.inlineScripts(config);
+        }else{
+          config.amdConfigFileName = amdConfigInfo.amdConfigFileName;
+          config.amdStartFileName = amdStartScriptInfo.amdStartFileName;
+          return self.replaceScripts(config);
+        }
+      })
+      .then(function(result){
+        return result;
+      });
+  },
+
+  /**
+   * If configured, read the amd config file into a string.
+   * If we are not inlining scripts, compute the hash, and 
+   * write to a fingerprinted file
+   * @param  {object} config Configuration object
+   * @return {object}        amdConfig object
+   */
+  handleAmdConfig: function(config){
+    var result = {
+      amdConfig: null,
+      amdConfigFileName:''
+    };
+    // If an amd config file is defined...
+    if (config.configPath) {
+      //read the file contents and cook a sha for it
+      result.amdConfig = fs.readFileSync(path.join(root, config.configPath), 'utf8');
+      //if we are not inlining...
+      if(!config.inline){
+        //fingerprint it and copy it to the output
+        var amdConfigSha = sha(result.amdConfig);
+        result.amdConfigFileName = 'assets/amd-config-' + amdConfigSha + '.js';
+        fs.writeFileSync(path.join(config.directory, result.amdConfigFileName), result.amdConfig);
+      }
+    }
+    return result;
+  },
+
+  /**
+   * Inline the amd-config and amd-start scripts
+   * @param  {Object} config Configuration for this function
+   * @return {Promise}        Promise
+   */
+  inlineScripts: function(config){
```

Comment:
I think the two function inlineScripts and replaceScripts are 99% the same. I would like to have a single function and have the 1% differentiated by a if statement.